include/zephyr/drivers/rtc/rtc_rv3028.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024 ANITRA system s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief User EEPROM/RAM API extension for the RV3028 RTC driver.
+ *
+ * This header exposes extended functionality of the RV3028 RTC driver. The extension is invokable
+ * from user mode threads.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_RTC_RV3028_H
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_RTC_RV3028_H
+
+#include <stdint.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Read a single byte into the user RAM register.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param addr RAM address to read from (0x00 or 0x01).
+ * @param val Pointer to the variable to store the read value.
+ *
+ * @retval 0 If successful.
+ * @retval -EINVAL Invalid address parameter.
+ * @retval -EIO General input / output error.
+ */
+__syscall int rv3028_ram_read(const struct device *dev, uint8_t addr, uint8_t *val);
```

Comment:
@bjarki-trackunit does it make sense to make a generic function out of it? Because some RTCs (like the RV-8263 and the DS1307) have internal RAM which can be used.

Reply:
Yes, that was my question too. Currently it is not possible for one device to implement multiple APIs, right?

Reply:
It's not supported by the RTC API. The RTC API can not use a second API (same for other device types) do you can not combine them. What you need are two functions in the API structure for reading and writing RAM.

Reply:
The solution is to expose the memory as a memory device, which is a child of the RTC. The memory device will then implement an appropriate API for memory access (may I suggest https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/retained_mem.h)

An example devicetree layout
```
rtc: rtc@2 {
        compatible: "microcrystal,rv3028";
        status = "okay";

        retmem: retmem {
                compatible: "microcrystal,rv3028-retmem";
                status = "okay";
        };
};
```
The rtc node implements the RTC API, the retmem node implements the retained_mem API. The device driver for the rv3028 defines both devices, both devices share the same data and config structures :)

This may be a bit overly complicated for this PR, I would add this in the future

Reply:
Ah this is a bummer 😕 I would prefer to be able to use the retention subsystem API or something like that. This does not only apply to RTC – many other devices contain retained registers. I think the RTC API should be limited to the time stuff.

Reply:
This problem is way larger than just RTCs, think of modems that contain a GNSS, cellular radio, SMS, and often whole flash filesystems, representing the "internal" hardware blocks of devices and modems as individual devices seems to be the only scalable way to fully expose the features of any device.

RTCs often contain watchdogs and low power counters as well, we can't support all of this in one API :)

Reply:
So we have to add an additional retmem driver to support this?